docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,298 @@
+learners to be enrolled in the Learning Path.
+
++---------------------+-------------------------------------------------------+
+| API Path            | /api/v1/learning-path-enrollment/<learning_path_id>   |
++---------------------+-------------------------------------------------------+
+| Methods             | GET, POST, DELETE                                     |
++---------------------+-------------------------------------------------------+
+| Query Params        | username (OPTIONAL)                                   |
++---------------------+-------------------------------------------------------+
+| Permissions Required| LoggedIn                                              |
++---------------------+-------------------------------------------------------+
+
+.. note::
+
+   The ``learning_path_id`` will be changed to a human readable content key.
+
+
+GET
+"""
+
+**Corresponding User Action:** Fetch enrollment in a Learning Path
+
+* When the ``username`` query parameter is present, the api returns
+
+  * the *active* enrollment object of the corresponding user for a **staff** user
+    or when both ``currentUser`` and the ``username`` point to the same user.
+  * HTTP 403 for a non-staff user, when ``currentUser`` and ``username``
+    refer to different users.
+
+* When the ``username`` query parameter is absent, the api returns
+
+  * all the *active* enrollments in a Learning Path if the ``currentUser`` is a **staff**
+    or **admin** user.
+  * the ``LearningPathEnrollment`` object if the logged-in user is enrolled in the
+    specified learning path. Otherwise returns a HTTP 404.
+
+
+POST
+""""
+
+**Corresponding User Action:** Enroll
+
+* When ``username`` is absent, enrolls the ``currentUser`` in the Learning
+  Path reference by the ``learning_path_id``.
+* When ``username`` is present
+
+  * ``username`` is different from the ``currentUser``, the user
+    linked to ``username`` will be enrolled, if the ``currentUser`` is a
+    **staff** or **admin**.
+  * Return an HTTP 403 error if the ``currentUser`` is a non-staff user
+  * Return an HTTP 404 if the ``username`` or ``learning_path_id`` do not
+    resolve into valid objects.
+
+* When an *active* ``LearningPathEnrollment`` object exists for the user and the
+  Learning Path, return HTTP 409 Conflict.
+
+
+DELETE
+""""""
+
+**Corresponding User Action:** Unenroll
+
+This will follow the same permission logic as the **POST** request allowing
+both self-unenrollment and staff-unenrollment.
+
+* Marks the *active* ``LearningPathEnrollment`` of the learner as inactive, by
+  setting the ``is_active`` field to ``False`` when the learner has an active
+  enrollment.
+* Return a 404 when there are no active enrollments for the learner.
+
+This method will be gated with a Django settings ``LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT``
+and ``LEARNING_PATHS_ALLOW_STAFF_UNENROLLMENT`` that will default to ``False`` and
```

Comment:
I believe we don't need `LEARNING_PATHS_ALLOW_STAFF_UNENROLLMENT`, and I'm not sure what the use case for this one would be. If there is a community use case for this, we can add it in the future.

Reply:
Wouldn't there be cases where learners drop out of a learning path like course and the staff has to unenroll them? I was thinking this flag would affect both the API and the Django Admin views.

Reply:
@tecoholic, ah, I meant that I don't see a use case where we would want to prevent all staff users from unenrolling learners.

There might be a use case where we want to allow some staff users to modify only Learning Path enrollments related to their organizations, but this would require a fine-grained RBAC, which is out of the scope here.